l10n_es_facturae/models/account_invoice.py

```diff
@@ -446,3 +468,59 @@ def _validate_facturae(self, xml_string):
                   "is the error, which may give you an idea on the cause "
                   "of the problem : %s") % str(e))
         return True
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    invoice_state = fields.Selection(
+        related='invoice_id.state',
+        readonly=True,
+    )
+    facturae_receiver_contract_reference = fields.Char()
+    facturae_receiver_contract_date = fields.Date()
+    facturae_receiver_transaction_reference = fields.Char()
+    facturae_receiver_transaction_date = fields.Date()
+    facturae_issuer_contract_reference = fields.Char()
+    facturae_issuer_contract_date = fields.Date()
+    facturae_issuer_transaction_reference = fields.Char()
+    facturae_issuer_transaction_date = fields.Date()
+    facturae_file_reference = fields.Char()
+    facturae_file_date = fields.Date()
+    facturae_start_date = fields.Date()
+    facturae_end_date = fields.Date()
+    facturae_transaction_date = fields.Date()
+
+    @api.constrains('facturae_start_date', 'facturae_end_date')
+    def _check_facturae_date(self):
+        for record in self:
+            if bool(record.facturae_start_date) != bool(
+                record.facturae_end_date
+            ):
+                raise ValidationError(_(
+                    'Start date is required if you define end date'
```

Comment:
El mensaje debería ser `FacturaE start and end dates are both required if one of them is filled.`